### PR TITLE
Added function translate to help info

### PR DESCRIPTION
**Pull request by areksolek**
Added function translate to help info

### CODE REVIEW

File: app/Language.php [hunk 1]
```diff
@@ -196,6 +196,32 @@ public static function translate($key, $moduleName = '_Base', $language = false,
...
+			$translated = \App\Language::translate($key, 'Other:HelpInfo');
+			if($key !== $translated){
+				return $translated;
+			}
+			$translated = \App\Language::translate($label, 'Other:HelpInfo');
+			if($label !== $translated){
+				return $translated;
+			}
+		}
+		return false;
```
Comment (rskrzypczak) [inline app/Language.php]: one return

File: layouts/basic/modules/KnowledgeBase/DetailViewBlockView.tpl [hunk 2]
```diff
@@ -44,11 +44,16 @@
...
-											{if in_array($VIEW,$HELPINFO) && \App\Language::translate($HELPINFO_LABEL, 'Other:HelpInfo') neq $HELPINFO_LABEL}
-												<a style="margin-left: 5px;margin-top: 2px;" href="#" class="js-help-info float-right" title="" data-placement="top" data-content="{\App\Language::translate($HELPINFO_LABEL, 'Other:HelpInfo')}" data-original-title='{\App\Language::translate($FIELD_MODEL->getFieldLabel(), $MODULE_NAME)}'><i class="fas fa-info-circle"></i></a>
-												{/if}
+											{assign var=HELPINFO_LABEL value=\App\Language::getTranslateHelpInfo($FIELD_MODEL,$VIEW)}
+											{if $HELPINFO_LABEL}
+												<a href="#" class="js-help-info float-right u-cursor-pointer"
+													title="" 
+													data-placement="top" 
+													data-content="{$HELPINFO_LABEL}" 
+													data-original-title='{$HELPINFO_LABEL}'>
```
Comment (rskrzypczak) [inline layouts/basic/modules/KnowledgeBase/DetailViewBlockView.tpl]: the same $HELPINFO_LABEL twice?

File: layouts/basic/modules/Vtiger/Detail/BlockView.tpl [hunk 3]
```diff
@@ -51,12 +51,17 @@
...
 										<label class="u-text-small-bold">
 											{\App\Language::translate({$FIELD_MODEL->getFieldLabel()},{$MODULE_NAME})}
-											{if in_array($VIEW,$HELPINFO) && \App\Language::translate($HELPINFO_LABEL, 'HelpInfo') neq $HELPINFO_LABEL}
-												<a href="#" class="js-help-info float-right u-cursor-pointer" title="" data-placement="top" data-content="{\App\Language::translate($HELPINFO_LABEL, 'HelpInfo')}" data-original-title='{\App\Language::translate($FIELD_MODEL->getFieldLabel(), $MODULE_NAME)}'><span class="fas fa-info-circle"></span></a>
+											{if $HELPINFO_LABEL}
+												<a href="#" class="js-help-info float-right u-cursor-pointer" 
+													title=""
+													data-placement="top" 
+													data-content="{$HELPINFO_LABEL}"
+													data-original-title='{$HELPINFO_LABEL}'>
```
Comment (rskrzypczak) [inline layouts/basic/modules/Vtiger/Detail/BlockView.tpl]: the same $HELPINFO_LABEL twice?
itd.

File: app/Language.php [hunk 4]
```diff
@@ -196,6 +196,31 @@ public static function translate($key, $moduleName = '_Base', $language = false,
...
+	 */
+	public static function getTranslateHelpInfo(\Vtiger_Field_Model $fieldModel, string $view)
+	{
+		$moduleName = $fieldModel->getModuleName();
+		$label = $fieldModel->getFieldLabel();
+		if(in_array($view,explode(',', $fieldModel->get('helpinfo')))){
+			$key = "$moduleName|$label";
+			$translated = \App\Language::translate($key, 'Other:HelpInfo');
+			if($key !== $translated){
+				return  $translated;
```
Comment (mariuszkrzaczkowski) [inline app/Language.php]: Phpcs

File: app/Language.php [hunk 5]
```diff
@@ -196,6 +196,31 @@ public static function translate($key, $moduleName = '_Base', $language = false,
...
+			}
+			$translated = \App\Language::translate($label, 'Other:HelpInfo');
+			if($label !== $translated){
+				return  $translated;
```
Comment (mariuszkrzaczkowski) [inline app/Language.php]: PHP cs

File: layouts/basic/modules/Vtiger/EditViewBlocks.tpl [hunk 6]
```diff
@@ -124,17 +124,17 @@
...
-												<a href="#" class="js-help-info float-right" title=""
-												   data-placement="top"
-												   data-content="{\App\Language::translate($HELPINFO_LABEL, 'Other:HelpInfo')}"
-												   data-original-title="{\App\Language::translate($FIELD_MODEL->getFieldLabel(), $MODULE)}">
+											{if $HELPINFO_LABEL}
+												<a href="#" class="js-help-info float-right u-cursor-pointer"
+													title=""
+													data-placement="top"
+													data-content="{$HELPINFO_LABEL}"
+													data-original-title='{\App\Language::translate($FIELD_MODEL->getFieldLabel(), $QUALIFIED_MODULE_NAME)}'>
```
Comment (rskrzypczak) [inline layouts/basic/modules/Vtiger/EditViewBlocks.tpl]: $QUALIFIED_MODULE_NAME ?

File: layouts/basic/modules/Vtiger/QuickCreate.tpl [hunk 7]
```diff
@@ -73,17 +73,17 @@
...
-													<a href="#" class="js-help-info float-right" title=""
-													   data-placement="top"
-													   data-content="{\App\Language::translate($HELPINFO_LABEL, 'Other:HelpInfo')}"
-													   data-original-title='{\App\Language::translate($FIELD_MODEL->getFieldLabel(), $MODULE)}'>
+												{if $HELPINFO_LABEL}
+													<a href="#" class="js-help-info float-right u-cursor-pointer"
+														title=""
+														data-placement="top"
+														data-content="{$HELPINFO_LABEL}"
+														data-original-title='{\App\Language::translate($FIELD_MODEL->getFieldLabel(), $MODULE)}'>
```
Comment (rskrzypczak) [inline layouts/basic/modules/Vtiger/QuickCreate.tpl]:
```suggestion
														data-original-title='{\App\Language::translate($FIELD_MODEL->getFieldLabel(), $MODULE_NAME)}'>
```

File: app/Language.php [hunk 8]
```diff
@@ -196,6 +196,30 @@ public static function translate($key, $moduleName = '_Base', $language = false,
 		return Purifier::encodeHtml($key);
 	}
 
+	/**
+	 * Functions get translate help info.
+	 *
+	 * @param \Vtiger_Field_Model $fieldModel
+	 * @param string              $view
+	 *
+	 * @return bool $view
```
Comment (rskrzypczak) [inline app/Language.php]:
```suggestion
	 * @return string
```

File: app/Language.php [hunk 9]
```diff
@@ -196,6 +196,30 @@ public static function translate($key, $moduleName = '_Base', $language = false,
...
+	 */
+	public static function getTranslateHelpInfo(\Vtiger_Field_Model $fieldModel, string $view)
+	{
+		$moduleName = $fieldModel->getModuleName();
```
Comment (rskrzypczak) [inline app/Language.php]: move to if

File: app/Language.php [hunk 10]
```diff
@@ -196,6 +196,30 @@ public static function translate($key, $moduleName = '_Base', $language = false,
...
+		$label = $fieldModel->getFieldLabel();
+		$translate = '';
+		if (\in_array($view, explode(',', $fieldModel->get('helpinfo')))) {
+			$key = "$moduleName|$label";
+			if (($translated = self::translate($key, 'Other:HelpInfo')) !== $key) {
+				$translate = $translated;
+			} elseif (($translated = self::translate($label, 'Other:HelpInfo')) !== $label) {
+				$translate = $translated;
+			}
+			return $translate;
```
Comment (rskrzypczak) [inline app/Language.php]: move to end function

File: layouts/basic/modules/Vtiger/Detail/Widget/BlockView.tpl [hunk 11]
```diff
@@ -34,16 +34,16 @@
...
-											{if in_array($VIEW,$HELPINFO) && \App\Language::translate($HELPINFO_LABEL, 'Other:HelpInfo') neq $HELPINFO_LABEL}
+											{if $HELPINFO_LABEL}
 												<a href="#" class="js-help-info float-right u-cursor-pointer"
-												   title="" data-placement="top"
-												   data-content="{\App\Language::translate($HELPINFO_LABEL, 'Other:HelpInfo')}"
-												   data-original-title='{\App\Language::translate($FIELD_MODEL->getFieldLabel(), $MODULE_NAME)}'>
+													title=""
+													data-placement="top"
+													data-content="{$HELPINFO_LABEL}"
+													data-original-title='{\App\Language::translate($FIELD_MODEL->getFieldLabel(), $MODULE_NAME)}'>
```
Comment (TomaszQr) [inline layouts/basic/modules/Vtiger/Detail/Widget/BlockView.tpl]: double quotes

File: layouts/basic/modules/Vtiger/Detail/BlockView.tpl [hunk 12]
```diff
@@ -51,12 +51,17 @@
...
 										<label class="u-text-small-bold">
 											{\App\Language::translate({$FIELD_MODEL->getFieldLabel()},{$MODULE_NAME})}
-											{if in_array($VIEW,$HELPINFO) && \App\Language::translate($HELPINFO_LABEL, 'HelpInfo') neq $HELPINFO_LABEL}
-												<a href="#" class="js-help-info float-right u-cursor-pointer" title="" data-placement="top" data-content="{\App\Language::translate($HELPINFO_LABEL, 'HelpInfo')}" data-original-title='{\App\Language::translate($FIELD_MODEL->getFieldLabel(), $MODULE_NAME)}'><span class="fas fa-info-circle"></span></a>
+											{if $HELPINFO_LABEL}
+												<a href="#" class="js-help-info float-right u-cursor-pointer"
+													title=""
+													data-placement="top"
+													data-content="{$HELPINFO_LABEL}"
+													data-original-title='{\App\Language::translate($FIELD_MODEL->getFieldLabel(), $MODULE_NAME)}'>
```
Comment (TomaszQr) [inline layouts/basic/modules/Vtiger/Detail/BlockView.tpl]: double quotes

File: layouts/basic/modules/KnowledgeBase/DetailViewBlockView.tpl [hunk 13]
```diff
@@ -44,11 +44,16 @@
...
-											{if in_array($VIEW,$HELPINFO) && \App\Language::translate($HELPINFO_LABEL, 'Other:HelpInfo') neq $HELPINFO_LABEL}
-												<a style="margin-left: 5px;margin-top: 2px;" href="#" class="js-help-info float-right" title="" data-placement="top" data-content="{\App\Language::translate($HELPINFO_LABEL, 'Other:HelpInfo')}" data-original-title='{\App\Language::translate($FIELD_MODEL->getFieldLabel(), $MODULE_NAME)}'><i class="fas fa-info-circle"></i></a>
-												{/if}
+											{assign var=HELPINFO_LABEL value=\App\Language::getTranslateHelpInfo($FIELD_MODEL,$VIEW)}
+											{if $HELPINFO_LABEL}
+												<a href="#" class="js-help-info float-right u-cursor-pointer"
+													title=""
+													data-placement="top"
+													data-content="{$HELPINFO_LABEL}"
+													data-original-title='{\App\Language::translate($FIELD_MODEL->getFieldLabel(), $MODULE_NAME)}'>
```
Comment (TomaszQr) [inline layouts/basic/modules/KnowledgeBase/DetailViewBlockView.tpl]: double quotes

File: layouts/basic/modules/Calendar/Extended/EventForm.tpl [hunk 14]
```diff
@@ -53,17 +53,17 @@
...
-												role="tooltip">
-												<span class="fas fa-info-circle fa-sm align-baseline"></span>
-											</a>
-										{/if}
+										{if $HELPINFO_LABEL}
+												<a href="#" class="js-help-info float-right u-cursor-pointer"
+													title=""
+													data-placement="top"
+													data-content="{$HELPINFO_LABEL}"
+													data-original-title='{\App\Language::translate($FIELD_MODEL->getFieldLabel(), $MODULE_NAME)}'>
```
Comment (TomaszQr) [inline layouts/basic/modules/Calendar/Extended/EventForm.tpl]: double quotes

File: layouts/basic/modules/Vtiger/Detail/Widget/GeneralInfo.tpl [hunk 15]
```diff
@@ -17,14 +17,16 @@
...
-											   data-original-title='{\App\Language::translate($FIELD_MODEL->getFieldLabel(), $MODULE_NAME)}'><span
-														class="fas fa-info-circle"></span></a>
-										{/if}
+											{assign var=HELPINFO_LABEL value=\App\Language::getTranslateHelpInfo($FIELD_MODEL,$VIEW)}
+										{if $HELPINFO_LABEL}
+												<a href="#" class="js-help-info float-right u-cursor-pointer"
+													title=""
+													data-placement="top"
+													data-content="{$HELPINFO_LABEL}"
+													data-original-title='{\App\Language::translate($FIELD_MODEL->getFieldLabel(), $MODULE_NAME)}'>
```
Comment (TomaszQr) [inline layouts/basic/modules/Vtiger/Detail/Widget/GeneralInfo.tpl]: double quotes

File: layouts/basic/modules/Vtiger/EditViewBlocks.tpl [hunk 16]
```diff
@@ -124,17 +124,17 @@
...
-												<a href="#" class="js-help-info float-right" title=""
-												   data-placement="top"
-												   data-content="{\App\Language::translate($HELPINFO_LABEL, 'Other:HelpInfo')}"
-												   data-original-title="{\App\Language::translate($FIELD_MODEL->getFieldLabel(), $MODULE)}">
+											{if $HELPINFO_LABEL}
+												<a href="#" class="js-help-info float-right u-cursor-pointer"
+													title=""
+													data-placement="top"
+													data-content="{$HELPINFO_LABEL}"
+													data-original-title='{\App\Language::translate($FIELD_MODEL->getFieldLabel(), $FIELD_MODEL->getModuleName())}'>
```
Comment (TomaszQr) [inline layouts/basic/modules/Vtiger/EditViewBlocks.tpl]: double quotes

File: layouts/basic/modules/Vtiger/QuickCreate.tpl [hunk 17]
```diff
@@ -73,17 +73,17 @@
...
-													<a href="#" class="js-help-info float-right" title=""
-													   data-placement="top"
-													   data-content="{\App\Language::translate($HELPINFO_LABEL, 'Other:HelpInfo')}"
-													   data-original-title='{\App\Language::translate($FIELD_MODEL->getFieldLabel(), $MODULE)}'>
+												{if $HELPINFO_LABEL}
+													<a href="#" class="js-help-info float-right u-cursor-pointer"
+														title=""
+														data-placement="top"
+														data-content="{$HELPINFO_LABEL}"
+														data-original-title='{\App\Language::translate($FIELD_MODEL->getFieldLabel(), $MODULE_NAME)}'>
```
Comment (TomaszQr) [inline layouts/basic/modules/Vtiger/QuickCreate.tpl]: double quotes

**Pull request merged**